linux/src/threadpool_linux.c

```diff
@@ -580,7 +626,31 @@ THANDLE(THREADPOOL_TIMER) threadpool_timer_start(THANDLE(THREADPOOL) threadpool,
                         /* Codes_SRS_THREADPOOL_LINUX_07_061: [ threadpool_timer_start shall return and allocated handle in timer_handle. ]*/
                         /* Codes_SRS_THREADPOOL_LINUX_07_062: [ threadpool_timer_start shall succeed and return a non-NULL handle. ]*/
                         result->time_id = time_id;
-                        goto all_ok;
+
+                        srw_lock_ll_acquire_exclusive(&timer_table_lock);
+                        uint32_t retry_times = 0;
+                        for (uint32_t i = 0; i < TIMER_TABLE_SIZE; i++)
+                        {
+                            if (timer_table[i] == NULL)
+                            {
+                                timer_table[i] = result;
+                                srw_lock_ll_release_exclusive(&timer_table_lock);
+                                goto all_ok;
+                            }
+                            else if (i == TIMER_TABLE_SIZE - 1)
+                            {
+                                if (retry_times < MAX_TIMER_START_RETRY)
+                                {
+                                    ThreadAPI_Sleep(10);
+                                    i = 0;
```

Comment:
This retry does not really solve anything, does it?
If we're out of timers, we're out of timers, we're probably pushing the system to what it can do and if we are pushing it it will fail after 1 or 2 retries also.

I feel having retries actually muddies the implementation because  it tries to hide some errors but in a rather unpredictable way ...

Reply:
removed the retry